orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py

```diff
@@ -277,14 +280,16 @@ def _initialize_graph_builder(self, training):
         grad_builder_config.initializer_names_to_train = initializer_names_to_train
         grad_builder_config.input_names_require_grad = self._input_info.require_grad_names
         grad_builder_config.build_gradient_graph = training
-        grad_builder_config.graph_transformer_config = C.GraphTransformerConfiguration()
-        grad_builder_config.graph_transformer_config.propagate_cast_ops_level = self._propagate_cast_ops_level
-        grad_builder_config.graph_transformer_config.propagate_cast_ops_allow = self._propagate_cast_ops_allow
+        grad_builder_config.graph_transformer_config = C.TrainingGraphTransformerConfiguration()
+        grad_builder_config.graph_transformer_config.propagate_cast_ops_config = C.PropagateCastOpsConfiguration()
+        grad_builder_config.graph_transformer_config.propagate_cast_ops_config.level = self._propagate_cast_ops_level
+        grad_builder_config.graph_transformer_config.propagate_cast_ops_config.allow = self._propagate_cast_ops_allow
+        grad_builder_config.graph_transformer_config.propagate_cast_ops_config.strategy = self._propagate_cast_ops_strategy
         grad_builder_config.graph_transformer_config.allow_layer_norm_mod_precision = self._allow_layer_norm_mod_precision
-        grad_builder_config.loglevel = {_logger.LogLevel.VERBOSE : C.Severity.VERBOSE,
-                                        _logger.LogLevel.INFO : C.Severity.INFO,
-                                        _logger.LogLevel.WARNING : C.Severity.WARNING,
-                                        _logger.LogLevel.ERROR : C.Severity.ERROR,
-                                        _logger.LogLevel.FATAL : C.Severity.FATAL}.get(self._loglevel, C.Severity.WARNING)
+        grad_builder_config.loglevel = {_logger.LogLevel.VERBOSE: C.Severity.VERBOSE,
+                                        _logger.LogLevel.INFO: C.Severity.INFO,
+                                        _logger.LogLevel.WARNING: C.Severity.WARNING,
+                                        _logger.LogLevel.ERROR: C.Severity.ERROR,
+                                        _logger.LogLevel.FATAL: C.Severity.FATAL}.get(self._loglevel, C.Severity.WARNING)
```

Comment:
one idea to get rid of this adhoc dict would be to create a helper function called `ortmodule_loglevel_to_onnxruntime_c_loglevel(loglevel)` in which given _logger loglevel, this method would return the correspoding C log level. That helper could be used in multiple places later